crates/stages/src/stages/execution.rs

```diff
@@ -106,16 +113,10 @@ impl<DB: Database> Stage<DB> for ExecutionStage {
         // get canonical blocks (num,hash)
         let canonical_batch = canonicals
             .walk(start_block)?
-            .take(BATCH_SIZE as usize) // TODO: commit_threshold
+            .take_while(|entry| entry.as_ref().map(|e| e.0 <= end_block).unwrap_or_default())
             .map(|i| i.map(BlockNumHash))
             .collect::<Result<Vec<_>, _>>()?;
```

Comment:
Would like to put this in a helper maybe on the trait

Reply:
can clean up all db methods in a separate PR

Reply:
opened https://github.com/paradigmxyz/reth/issues/702